src/sentry/static/sentry/app/components/dropdownAutoComplete.jsx

```diff
@@ -31,7 +35,11 @@ class DropdownAutoComplete extends React.Component {
             <Actor
               isOpen={renderProps.isOpen}
               role="button"
-              onClick={renderProps.actions.open}
+              onClick={
+                renderProps.isOpen && shouldToggle
```

Comment:
I guess I'm struggling to see the impact of the `shouldToggle` flag, since it is always set to true. 

It seems like the switch on `renderProps.isOpen` would be enough?

Reply:
It defaults to false, we only set it to true for project and env selector. We use DropdownAutoComplete in a few other places.

Reply:
maybe it's just the name that is confusing me then.

Reply:
And since it could only open it before, this enables the close functionality only _if_ that flag is added. got it